src/share/man/man4/ti.4
.\"	ti.4,v 1.12 2013/07/20 21:39:58 wiz Exp
.\"
.\" Copyright (c) 1997, 1998, 1999
.\"	Bill Paul <wpaul@ctr.columbia.edu>. All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. All advertising materials mentioning features or use of this software
.\"    must display the following acknowledgement:
.\"	This product includes software developed by Bill Paul.
.\" 4. Neither the name of the author nor the names of any co-contributors
.\"    may be used to endorse or promote products derived from this software
.\"   without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY Bill Paul AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL Bill Paul OR THE VOICES IN HIS HEAD
.\" BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
.\" CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
.\" SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
.\" INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
.\" CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
.\" ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
.\" THE POSSIBILITY OF SUCH DAMAGE.
.\"
.\" FreeBSD Id: src/share/man/man4/ti.4,v 1.5 2000/01/23 16:13:24 asmodai Exp
.\"
.Dd June 2, 2001
.Dt TI 4
.Os
.Sh NAME
.Nm ti
.Nd
Alteon Networks Tigon I and Tigon II gigabit
.Tn Ethernet
driver
.Sh SYNOPSIS
.Cd "ti* at pci? dev ? function ?"
.Sh DESCRIPTION
The
.Nm
driver provides support for PCI gigabit
.Tn Ethernet
adapters based on the Alteon Networks Tigon gigabit
.Tn Ethernet
controller chip.
The Tigon contains an embedded R4000 CPU, gigabit MAC, dual DMA channels
and a PCI interface unit.
The Tigon II contains two R4000 CPUs and other refinements.
Either chip can be used in either a 32-bit or 64-bit PCI slot.
Communication with the chip is achieved via PCI shared memory
and bus master DMA.
The Tigon I and II support hardware multicast address filtering,
VLAN tag extraction and insertion, and jumbo
.Tn Ethernet
frames sizes up to 9000 bytes.
Note that the Tigon I chipset is no longer in active production:
all new adapters should come equipped with Tigon II chipsets.
.Pp
There are several PCI boards available from both Alteon and other
vendors that use the Tigon chipset under OEM contract.
The
.Nm
driver has been tested with the following Tigon-based adapters:
.Bl -bullet -offset indent
.It
The Alteon AceNIC V gigabit (1000BASE-SX and 1000BASE-T variants)
.Tn Ethernet
adapter
.It
The 3Com 3c985-SX gigabit
.Tn Ethernet
adapter
.It
The Netgear GA620 gigabit (1000BASE-SX and 1000BASE-T variants)
.Tn Ethernet
adapter
.It
The Digital EtherWORKS 1000SX PCI Gigabit Adapter (DEGPA)
.El
.Pp
The following should also be supported but have not yet been tested:
.Bl -bullet -offset indent
.It
Silicon Graphics PCI gigabit
.Tn Ethernet
adapter
.El
.Pp
While the Tigon chipset supports 10, 100 and 1000Mbps speeds, support for
10 and 100Mbps speeds is only available on boards with the proper
transceivers.
Most adapters are only designed to work at 1000Mbps, however the driver
should support those NICs that work at lower speeds as well.
.Pp
Support for jumbo frames is provided via the interface MTU setting.
Selecting an MTU larger than 1500 bytes with the
.Xr ifconfig 8
utility configures the adapter to receive and transmit jumbo frames.
Using jumbo frames can greatly improve performance for certain tasks,
such as file transfers and data streaming.
.Pp
The
.Nm
driver supports the following media types:
.Pp
.Bl -tag -width xxxxxxxxxxxxxxxxxxxx
.It autoselect
Enable autoselection of the media type and options.
.It 10baseT/UTP
Set 10Mbps operation.
The
.Ar mediaopt
option can also be used to select either
.Ar full-duplex
or
.Ar half-duplex modes .
.It 100baseTX
Set 100Mbps (fast
.Tn Ethernet )
operation.
The
.Ar mediaopt
option can also be used to select either
.Ar full-duplex
or
.Ar half-duplex
modes.
.It 1000baseSX
Set 1000Mbps (gigabit
.Tn Ethernet over multimode fiber ) operation.
Only full
.Ar full-duplex
mode is supported at this speed.
.It 1000baseT
Set 1000Mbps (gigabit
.Tn Ethernet over twisted pair ) operation.
Only full
.Ar full-duplex
mode is supported at this speed.
.El
.Pp
The
.Nm
driver supports the following media options:
.Pp
.Bl -tag -width xxxxxxxxxxxxxxxxxxxx
.It full-duplex
Force full duplex operation.
.It half-duplex
Force half duplex operation.
.El
.Pp
The Alteon Tigon and Tigon II support IPv4/TCP/UDP checksumming in
hardware.
The
.Nm
supports this feature of the chip's firmware.
See
.Xr ifconfig 8
for information on how to enable this feature.
.Pp
For more information on configuring this device, see
.Xr ifconfig 8 .
.Sh DIAGNOSTICS
.Bl -diag
.It "ti%d: can't map memory space"
A fatal initialization error has occurred.
.It "ti%d: couldn't map / establish interrupt"
A fatal initialization error has occurred.
.It "ti%d: jumbo buffer allocation failed"
The driver failed to allocate memory for jumbo frames during
initialization.
.It "ti%d: bios thinks we're in a 64 bit slot, but we aren't"
The BIOS has programmed the NIC as though it had been installed in
a 64-bit PCI slot, but in fact the NIC is in a 32-bit slot.
This happens as a result of a bug in some BIOSes.
This can be worked around on the Tigon II, but on the Tigon I
initialization will fail.
.It "ti%d: board self-diagnostics failed!"
The ROMFAIL bit in the CPU state register was set after system
startup, indicating that the on-board NIC diagnostics failed.
.It "ti%d: unknown hwrev"
The driver detected a board with an unsupported hardware revision.
The
.Nm
driver supports revision 4 (Tigon 1) and revision 6 (Tigon 2) chips
and has firmware only for those devices.
.It "ti%d: watchdog timeout"
The device has stopped responding to the network, or there is a problem with
the network connection (cable).
.El
.Sh SEE ALSO
.Xr netintro 4 ,
.Xr pci 4 ,
.Xr ifconfig 8
.Sh HISTORY
The
.Nm
device driver first appeared in
.Nx 1.4.2 .
.Sh AUTHORS
The
.Nm
driver was written by
.An Bill Paul Aq Mt wpaul@ctr.columbia.edu .
.Sh BUGS
The driver currently tries to access some on-board memory transparently.
This mapping (BUS_SPACE_MAP_LINEAR) fails on systems where the corresponding
PCI memory range is located in "sparse" space only.
.Pp
This driver currently does not work on big-endian systems.


src/lib/libterminfo/ti.c
/* ti.c,v 1.3 2013/06/07 13:16:18 roy Exp */

/*
 * Copyright (c) 2009 The NetBSD Foundation, Inc.
 *
 * This id is derived from software contributed to The NetBSD Foundation
 * by Roy Marples.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source id must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
 * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
 * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
 * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
 * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
 * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

#include <sys/cdefs.h>
__RCSID("ti.c,v 1.3 2013/06/07 13:16:18 roy Exp");

#include <assert.h>
#include <string.h>
#include <term_private.h>
#include <term.h>

int
ti_getflag(const TERMINAL *term, const char *id)
{
	ssize_t ind;
	size_t i;
	TERMUSERDEF *ud;

	_DIAGASSERT(term != NULL);
	_DIAGASSERT(id != NULL);

	ind = _ti_flagindex(id);
	if (ind != -1)
		return term->flags[ind];
	for (i = 0; i < term->_nuserdefs; i++) {
		ud = &term->_userdefs[i];
		if (ud->type == 'f' && strcmp(ud->id, id) == 0)
			return ud->flag;
	}
	return ABSENT_BOOLEAN;
}

int
tigetflag(const char *id)
{

	_DIAGASSERT(id != NULL);
	if (cur_term != NULL)
		return ti_getflag(cur_term, id);
	return ABSENT_BOOLEAN;
}

int
ti_getnum(const TERMINAL *term, const char *id)
{
	ssize_t ind;
	size_t i;
	TERMUSERDEF *ud;

	_DIAGASSERT(term != NULL);
	_DIAGASSERT(id != NULL);

	ind = _ti_numindex(id);
	if (ind != -1) {
		if (!VALID_NUMERIC(term->nums[ind]))
			return ABSENT_NUMERIC;
		return term->nums[ind];
	}
	for (i = 0; i < term->_nuserdefs; i++) {
		ud = &term->_userdefs[i];
		if (ud->type == 'n' && strcmp(ud->id, id) == 0) {
			if (!VALID_NUMERIC(ud->num))
			    return ABSENT_NUMERIC;
			return ud->num;
		}
	}
	return CANCELLED_NUMERIC;
}

int
tigetnum(const char *id)
{

	_DIAGASSERT(id != NULL);
	if (cur_term != NULL)
		return ti_getnum(cur_term, id);
	return CANCELLED_NUMERIC;
}

const char *
ti_getstr(const TERMINAL *term, const char *id)
{
	ssize_t ind;
	size_t i;
	TERMUSERDEF *ud;

	_DIAGASSERT(term != NULL);
	_DIAGASSERT(id != NULL);

	ind = _ti_strindex(id);
	if (ind != -1)
		return term->strs[ind];
	for (i = 0; i < term->_nuserdefs; i++) {
		ud = &term->_userdefs[i];
		if (ud->type == 's' && strcmp(ud->id, id) == 0)
			return ud->str;
	}
	return (const char *)CANCELLED_STRING;
}

char *
tigetstr(const char *id)
{

	_DIAGASSERT(id != NULL);
	if (cur_term != NULL)
		return __UNCONST(ti_getstr(cur_term, id));
	return (char *)CANCELLED_STRING;
}
